File: plugins/claude-md-plugin/core/tests/fixtures/go/state_machine.go
package state

// State represents the state machine states.
type State int

const (
	StateIdle State = iota
	StateLoading
	StateLoaded
	StateError
)

// StateContext holds the state machine context.
type StateContext struct {
	State State
	Data  interface{}
	Error error
}

// ResourceLoader manages resource loading with lifecycle methods.
type ResourceLoader struct {
	context StateContext
}

// NewResourceLoader creates a new ResourceLoader.
func NewResourceLoader() *ResourceLoader {
	return &ResourceLoader{
		context: StateContext{State: StateIdle},
	}
}

// Init initializes the loader.
// @lifecycle 1
func (r *ResourceLoader) Init() {
	r.context = StateContext{State: StateIdle}
}

// Start begins loading resources.
// @lifecycle 2
func (r *ResourceLoader) Start() error {
	if r.context.State != StateIdle {
		return ErrInvalidState
	}
	r.context.State = StateLoading
	return nil
}

// Stop stops the loader.
// @lifecycle 3
func (r *ResourceLoader) Stop() {
	r.context.State = StateIdle
	r.context.Data = nil
}

// Destroy cleans up resources.
// @lifecycle 4
func (r *ResourceLoader) Destroy() {
	r.Stop()
}

// Load transitions from Idle to Loading.
func (r *ResourceLoader) Load() {
	r.context.State = StateLoading
}

// OnSuccess transitions from Loading to Loaded.
func (r *ResourceLoader) OnSuccess(data interface{}) {
	r.context.State = StateLoaded
	r.context.Data = data
}

// OnError transitions from Loading to Error.
func (r *ResourceLoader) OnError(err error) {
	r.context.State = StateError
	r.context.Error = err
}

// Retry transitions from Error to Idle.
func (r *ResourceLoader) Retry() {
	if r.context.State == StateError {
		r.context.State = StateIdle
		r.context.Error = nil
	}
}

var ErrInvalidState = errors.New("invalid state")

import "errors"

File: plugins/claude-md-plugin/core/tests/fixtures/python/state_machine.py
"""State machine example for Python code analyzer."""

from enum import Enum
from typing import Optional, Any


class State(Enum):
    """State machine states."""
    IDLE = "idle"
    LOADING = "loading"
    LOADED = "loaded"
    ERROR = "error"


class StateContext:
    """Context for the state machine."""

    def __init__(self):
        self.state: State = State.IDLE
        self.data: Optional[Any] = None
        self.error: Optional[Exception] = None


class ResourceLoader:
    """
    Resource loader with explicit lifecycle methods.

    Lifecycle:
        1. init() - Initialize the loader
        2. start() - Start loading resources
        3. stop() - Stop the loader
        4. destroy() - Clean up resources
    """

    def __init__(self):
        self._context = StateContext()

    def init(self) -> None:
        """
        Initialize the loader.

        @lifecycle 1
        """
        self._context = StateContext()

    def start(self) -> None:
        """
        Start loading resources.

        @lifecycle 2
        """
        if self._context.state != State.IDLE:
            raise RuntimeError("Can only start from IDLE state")
        self._context.state = State.LOADING

    def stop(self) -> None:
        """
        Stop the loader.

        @lifecycle 3
        """
        self._context.state = State.IDLE
        self._context.data = None

    def destroy(self) -> None:
        """
        Clean up resources.

        @lifecycle 4
        """
        self.stop()
        # Clean up...

    def load(self) -> None:
        """State transition: IDLE -> LOADING"""
        self._context.state = State.LOADING

    def on_success(self, data: Any) -> None:
        """State transition: LOADING -> LOADED"""
        self._context.state = State.LOADED
        self._context.data = data

    def on_error(self, error: Exception) -> None:
        """State transition: LOADING -> ERROR"""
        self._context.state = State.ERROR
        self._context.error = error

    def retry(self) -> None:
        """State transition: ERROR -> IDLE (retry)"""
        if self._context.state == State.ERROR:
            self._context.state = State.IDLE
            self._context.error = None
